frontend/packages/dev-console/src/components/pipelines/pipeline-overview/PipelineStartButton.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Button } from '@patternfly/react-core';
+import { useAccessReview } from '@console/internal/components/utils';
+import { AccessReviewResourceAttributes } from '@console/internal/module/k8s';
+import { impersonateStateToProps } from '@console/internal/reducers/ui';
+import { PipelineRunModel } from '../../../models';
+import { Pipeline } from '../../../utils/pipeline-augment';
+import { startPipelineModal } from '../modals';
+
+type StateProps = {
+  kind: string;
+  name: string;
+  subprotocols: string[];
+};
+
+type PipelineStartButtonProps = {
+  pipeline: Pipeline;
+  namespace: string;
+  impersonate?: StateProps;
+};
+
+const PipelineStartButton: React.FC<PipelineStartButtonProps> = ({
```

Comment:
So sorry, this is not what I meant. See below:

```suggestion
type StateProps = {
  impersonate?: {
    kind: string;
    name: string;
    subprotocols: string[];
  };
};

type PipelineStartButtonProps = {
  pipeline: Pipeline;
  namespace: string;
};

const PipelineStartButton: React.FC<PipelineStartButtonProps & StateProps> = ({
```

Reply:
This way the reader of the file can tell what they need to provide and what is provided by the connect.

Reply:
Thanks for clarifying. Updated.